File: VIF_git/GSDS-Qual/LeetCode/week1/Linked_List/Delete_the_Middle_Node_of_a_Linked_List/sunghyun.py
import math
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteMiddle(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        lllen = 0

        curptr = head

        while(curptr != None):
            lllen = lllen + 1
            curptr = curptr.next
        
        delind = math.floor(lllen / 2)

        if (delind == 0):
            return None

        curptr = head

        while(delind > 1):
            curptr = curptr.next
            delind = delind - 1

        curptr.next = curptr.next.next

        return head
        

        

File: VIF_git/GSDS-Qual/LeetCode/week1/Linked_List/Delete_the_Middle_Node_of_a_Linked_List/sunghyun.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteMiddle(struct ListNode* head) {
    if(head->next == NULL)
        return NULL;

    struct ListNode * temp = head;
    int count = 0;

    while(temp != NULL){
        count++;
        temp = temp->next;
    }

    temp = head;

    for(int i = 0; i < count/2 - 1; i++){
        temp = temp->next;
    }

    temp->next = temp->next->next;
    return head;
}